components/catalogs/front/src/components/ProductSelection/hooks/useCategories.ts

```diff
@@ -0,0 +1,43 @@
+import {useQuery} from 'react-query';
+import {useUserContext} from '@akeneo-pim-community/shared';
+import {Category, CategoryCode} from '../models/Category';
+
+type QueryParams = {
+    codes?: CategoryCode[];
+    isRoot?: boolean;
+};
+type Data = Category[];
+type ResultError = Error | null;
+type Result = {
+    isLoading: boolean;
+    isError: boolean;
+    data: Data | undefined;
+    error: ResultError;
+};
+
+export const useCategories = ({codes = [], isRoot = false}: QueryParams): Result => {
+    const locale = useUserContext().get('uiLocale');
```

Comment:
```suggestion
    const locale = useUserContext().get('catalogLocale');
```

To translate the data, it's `catalogLocale`, not `uiLocale`.